### PR TITLE
[runtime] add cpu device for PySession

### PR DESCRIPTION
**Pull request by YellowHCH**
* add cpu device in pysession
* add simple mlir test case

### CODE REVIEW

File: compiler/python/byteir/compile.py [hunk 1]
```diff
@@ -264,6 +264,11 @@ def _compile_cpu(
     with open(output_file_path, "w") as f:
         f.write(module.operation.get_asm())
 
+register_byteir_compiler_backend(compiler_fn=_compile_cuda, target="cuda", device="cuda")
```
Comment (qingyunqu) [inline compiler/python/byteir/compile.py:267]: Using decorator is more Pythonic.

File: compiler/python/CMakeLists.txt [hunk 2]
```diff
@@ -17,6 +17,7 @@ declare_mlir_python_sources(ByteIRPythonSources.Common
   SOURCES
     __init__.py
     _byteir_registry.py
+    registry.py
```
Comment (qingyunqu) [inline compiler/python/CMakeLists.txt:20]: `registry.py` => `_backend_registry.py`

File: compiler/python/byteir/compile.py (hunk 1 above)
Comment (YellowHCH) [reply compiler/python/byteir/compile.py:267]: > Using decorator is more Pythonic.

We need to consider that one could register different targets for the same `compiler_fn`, just like `_compile_cuda_with_ait`.

Comment (qingyunqu) [reply compiler/python/byteir/compile.py:267]: I think split `_compile_cuda_with_ait` into two functions `_compile_cuda_with_ait` and `_compile_cuda_with_ait_aggressive` is ok.

Comment (YellowHCH) [reply compiler/python/byteir/compile.py:267]: > I think split `_compile_cuda_with_ait` into two functions `_compile_cuda_with_ait` and `_compile_cuda_with_ait_aggressive` is ok.

That's great.

**Review by qingyunqu — APPROVED** [review]
LGTM

**Pull request merged**